worker/tls.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"math/big"
+	"net"
+	"os"
+	"time"
+
+	"github.com/koraygocmen/gravitational/certs"
+)
+
+// checkKeyCrt checks if the crt and key file exists
+// on the specified path, if not it will generate a new
+// crt and key file.
+func checkKeyCrt() {
+	if _, err := os.Stat(grpcServerCrtFile); err != nil {
+		if os.IsNotExist(err) {
+			crt := certs.CrtTemplate{
+				Organization: []string{"Gravitational"},
+				IP:           []net.IP{net.ParseIP("127.0.0.1")},
+				SerialNumber: big.NewInt(1),
+				NotBefore:    time.Now(),
+				NotAfter:     time.Now().Add(24 * 365 * time.Hour),
+				IsCa:         true,
```

Comment:
The worker really isn't a CA here? I think what you really want to do here is create a `tls.Config` and provide it with the certificate of the CA you generate on the scheduler.

Reply:
I agree, it looks like the way it's implemented now the workers will get "x509 signed by unrecognized authority" error when connecting to the scheduler, unless you add its CA to the system trust store?

Reply:
So it looks like this certificate isn't used at all in the current code. The worker appears to be loading the scheduler cert for the gRPC client, and despite it not having client ExtKeyUsage, it looks like it's being accepted.

Reply:
Yes, this currently is not used. When I implement the worker grpc server it will present this crt and key to allow connections. Sorry for the confusion